rpd_generator/bdl_structure/bdl_commands/system.py

```diff
@@ -1330,6 +1394,73 @@ def populate_air_energy_recovery(self):
                 self.air_energy_recovery_outdoor_airflow
             )
 
+    def populate_fan_operation_during_occupied(self):
+        fan_sch_name = self.keyword_value_pairs.get(BDL_SystemKeywords.FAN_SCHEDULE)
+
+        if (
+            self.keyword_value_pairs.get(BDL_SystemKeywords.TYPE)
+            == BDL_SystemTypes.RESVVT
+        ):
+            return FanSystemOperationOptions.CYCLING
+
+        elif (
+            self.keyword_value_pairs.get(BDL_SystemKeywords.TYPE)
+            == BDL_SystemTypes.DOAS
+        ):
+            fan_sch = self.rmd.bdl_obj_instances.get(fan_sch_name)
+            if not fan_sch:
+                return FanSystemOperationOptions.INTERMITTENT
+            else:
+                return FanSystemOperationOptions.CONTINUOUS
+
+        elif self.keyword_value_pairs.get(BDL_SystemKeywords.TYPE) in [
+            BDL_SystemTypes.PTAC,
+            BDL_SystemTypes.HP,
+            BDL_SystemTypes.UVT,
+        ]:
+            if not self.fan_sys_minimum_outdoor_airflow:
+                return FanSystemOperationOptions.CYCLING
+
+            min_oa_sch_name = self.keyword_value_pairs.get(
+                BDL_SystemKeywords.MIN_AIR_SCH
+            )
+            fan_sch = self.rmd.bdl_obj_instances.get(fan_sch_name)
+            min_oa_sch = self.rmd.bdl_obj_instances.get(min_oa_sch_name)
+            if not min_oa_sch:
+                return FanSystemOperationOptions.CONTINUOUS
+            if not fan_sch:  # (and min_oa_sch)
+                min_oa_sch_values = min_oa_sch.hourly_values
+                if min_oa_sch_values:
+                    for i in range(len(min_oa_sch_values)):
+                        if min_oa_sch_values[i] == 0:
+                            return FanSystemOperationOptions.CYCLING
+                    return FanSystemOperationOptions.CONTINUOUS
+                return FanSystemOperationOptions.CYCLING
+            if fan_sch and min_oa_sch:
+                fan_sch_values = fan_sch.hourly_values
+                min_oa_sch_values = min_oa_sch.hourly_values
+                if fan_sch_values and min_oa_sch_values:
+                    for i in range(len(fan_sch_values)):
+                        if min_oa_sch_values[i] == 0 and fan_sch_values[i] in (1, -999):
+                            return FanSystemOperationOptions.CYCLING
+                    return FanSystemOperationOptions.CONTINUOUS
+
+        elif self.keyword_value_pairs.get(BDL_SystemKeywords.TYPE) in [
+            BDL_SystemTypes.PSZ,
+            BDL_SystemTypes.PVVT,
+            BDL_SystemTypes.RESYS2,
+            BDL_SystemTypes.EVAP_COOL,
+            BDL_SystemTypes.FC,
+            BDL_SystemTypes.SZRH,
+            BDL_SystemTypes.HP,
+        ]:
+            return self.occupied_fan_operation_map.get(
+                self.keyword_value_pairs.get(BDL_SystemKeywords.INDOOR_FAN_MODE)
```

Comment:
Need to account for schedules with all 0s and I think some testing needs to be completed with -999. That is a real wildcard because then we don't know what to consider occ hours from the schedule.

Reply:
One idea is to have this be something special that the interface helps with - if a fan schedule has a -999 value in it then a window pops up and asks if -999 values occur during occupied hours, unoccupied hours, or both.

Another thought is that since we observe odd behavior with -999 and the help text calls it out as being unsupported in DOE 2.3 we could try to further enforce that schedules do not contain any -999 by raising a validation error.

Reply:
Are you suggesting this for fan schedules only? If so, I think I like the idea of a validation error with an informative message to help the modeler fix the problema.

Reply:
Yes for fan schedules only, and even more specifically the error would only be raised for systems types NOT in this list:
BDL_SystemTypes.DOAS,
BDL_SystemTypes.PTAC,
BDL_SystemTypes.HP,
BDL_SystemTypes.UVT,
BDL_SystemTypes.UHT,
BDL_SystemTypes.PSZ,
BDL_SystemTypes.PVVT,
BDL_SystemTypes.RESYS2,
BDL_SystemTypes.EVAP_COOL,
BDL_SystemTypes.FC,
BDL_SystemTypes.SZRH,
BDL_SystemTypes.HP